src/screens/AddEdit/index.tsx

```diff
@@ -1,58 +1,176 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { formLabels, modes } from "../StringsHelper";
+import {createContact, updateContact} from "../../redux/actions/ActionCreators";
+import Groups from "../Groups/Groups";
 
-interface Props {
-    navigation: {
-        goBack: () => {},
-        navigate: (screenName: string, params?: object) => {},
-    },
-    route,
-}
-
-interface State {
-    id: number,
-    mode: string,
-}
-
-class AddEditScreen extends Component<Props, State> {
-    constructor(props: Props) {
-        super(props);
-        this.state = {
-            id: 0,
-            mode: '',
+const AddEditScreen  = ({route, navigation}) => {
+    //wartości początkowe
+    const {navigate} = useNavigation();
+    const {id, mode} = route.params;
+    const contacts = useSelector(getContacts);
+    const dispatch = useDispatch();
+    const [exampleInitialValue, setExampleInitialValue] = useState(0);
+    const isEdit = mode === modes.edit;
+    const [firstName, setFirstName] = useState( isEdit? contacts[id].firstName : '');
+    const [secondName, setSecondName] = useState(isEdit ? contacts[id].secondName : '');
+    const [lastName, setSurname] = useState(isEdit ? contacts[id].lastName : '');
+    const [numbers, setNumbers] = useState(isEdit ? (contacts[id].telNumbers) : ([{number: '', category: '',}]));
+    const [deletedNumber, setDeletedNumber] = useState({index: -1, delNumber: {number: '', category: ''}});
+    const [emails, setEmails] = useState(isEdit ? (contacts[id].emails) : ([{email: '', category: '',}]));
+    const [deletedEmail, setDeletedEmail] = useState({index: -1, delEmail: {email: '', category: ''}});
+    const [snackbar, setSnackbar] = useState({isVisible: false, message: '', isActionVisible: false, label: ''});
+    const [isDeleteClicked, setIsDeleteClicked] = useState(false); //Logika pomagająca przy procesie usuwania/cofania usunięcia,
+    const buildContactObject = () => {
+        return  {
+            id: exampleInitialValue + 1,
+            firstName: firstName,
+            secondName: secondName,
+            lastName: lastName,
+            photoUrl: 'https://i.ytimg.com/vi/e5kVnW7E2YM/maxresdefault.jpg',
+            telNumbers: numbers,
+            emails: emails,
         };
-    }
-
-    onGroups = (id: number) => this.props.navigation.navigate('Groups', {id: id});
-
-    componentDidMount(): void {
-        this.setState({
-            mode: this.props.route.params.mode,
-            id: this.props.route.params.id,
-        });
-    }
-
-    render() {
-        const {
-            onGroups,
-        } = this;
-
-        const {
-            id,
-            mode,
-        } = this.state;
-
-        const {
-
-        } = this.props;
-
-        return (
-            <AddEdit
-                id={id}
-                mode={mode}
-                onGroups={onGroups}
-            />
-        );
-    }
-}
+    };
+    const contact = buildContactObject();
```

Comment:
Jeżeli chcesz już teraz to można użyć `const contact = useMemo(() => {tutaj_kod_z_buildContactObject}, [dependency]);` wtedy można na spokojnie do contactu się odwoływać.